### PR TITLE
heroku_addon_attachment

### LINKED ISSUE #3: Heroku: Attach addon to multiple apps
_This issue was originally opened by @Pluto1010 as hashicorp/terraform#10350. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

Feature-Request

I'm using Terraform v0.7.13

I think there is a function missing regarding the Heroku provider that enables me to express in a configfile that two apps share one resource. 

On Heroku you can attach some existing addons to multiple apps. "heroku addons:attach ...".

I want to use this to make use of a message broker addon. e.G. cloudamqp

### PR DESCRIPTION
**Pull request by JamesBelchamber**
This creates the heroku_addon_attachment resource, which allows pre-existing addons to be attached to additional apps. Fixes #3

### CODE REVIEW

File: heroku/resource_heroku_addon_attachment_test.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	})
+}
+
+func testAccCheckHerokuAddonAttachmentConfig_basic(appName string, addonName string) string {
+	return fmt.Sprintf(`
+resource "heroku_addon_attachment" "foobar" {
+    app_id   = "%s"
+    addon_id = "%s"
+}`, appName, addonName)
+}
```
Comment (tombuildsstuff) [inline heroku/resource_heroku_addon_attachment_test.go]: could we also add an additional test making use of the `name` field?

**Comment by JamesBelchamber** [comment]
The tests were a bit dumb anyway, I've refactored them and added an extra one :)

File: heroku/resource_heroku_addon_attachment_test.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+func testAccCheckHerokuAddonAttachmentConfig_named(appName string, addonName string, name string) string {
+	return fmt.Sprintf(`
+resource "heroku_addon_attachment" "foobar" {
+    app_id   = "%s"
```
Comment (tombuildsstuff) [inline heroku/resource_heroku_addon_attachment_test.go]: Thanks for adding the new test - I've just tried running the tests and they currently fail on an empty subscription due to the missing Heroku App:

```
$ acctests heroku TestAccHerokuAddonAttachment_
=== RUN   TestAccHerokuAddonAttachment_Basic
--- FAIL: TestAccHerokuAddonAttachment_Basic (1.40s)
	testing.go:434: Step 0 error: Error applying: 1 error(s) occurred:

		* heroku_addon_attachment.foobar: 1 error(s) occurred:

		* heroku_addon_attachment.foobar: Post https://api.heroku.com/addon-attachments: No such app.
=== RUN   TestAccHerokuAddonAttachment_Named
--- FAIL: TestAccHerokuAddonAttachment_Named (0.19s)
	testing.go:434: Step 0 error: Error applying: 1 error(s) occurred:

		* heroku_addon_attachment.foobar: 1 error(s) occurred:

		* heroku_addon_attachment.foobar: Post https://api.heroku.com/addon-attachments: No such app.
FAIL
exit status 1
FAIL	github.com/terraform-providers/terraform-provider-heroku/heroku	1.617s
```

.. could we update this to create the Heroku App as part of the tests so the tests are standalone? However this otherwise LGTM 👍

Comment (JamesBelchamber) [reply heroku/resource_heroku_addon_attachment_test.go]: Hmm. I haven't been running `acctest` so I didn't realise that. I'll get it working, will probably need to create an addon too.

**Comment by JamesBelchamber** [comment]
That should work better, we're creating the app and the addon now :)

File: heroku/resource_heroku_addon_attachment_test.go (hunk 2 above)
Comment (tombuildsstuff) [reply heroku/resource_heroku_addon_attachment_test.go]: @JamesBelchamber no worries, for context from the snippet above I'm using `acctests` as a bash wrapper for `TF_ACC=1 go test ./heroku ...` as it's less typing :)

Comment (tombuildsstuff) [reply heroku/resource_heroku_addon_attachment_test.go]: sure - this command will run just a single test (the test name is a prefix match):

```
TF_ACC=1 go test ./heroku -v -timeout 120m -run=TestAccHerokuAddonAttachment_Basic
```

hope that helps - let me know if you want me to run any of the tests to confirm too :)

**Comment by JamesBelchamber** [comment]
```
$ TF_ACC=1 go test ./heroku -v -timeout 120m -run=TestAccHerokuAddonAttachment_Basic
=== RUN   TestAccHerokuAddonAttachment_Basic
--- PASS: TestAccHerokuAddonAttachment_Basic (6.44s)
PASS
ok      github.com/terraform-providers/terraform-provider-heroku/heroku 6.453s
$ TF_ACC=1 go test ./heroku -v -timeout 120m -run=TestAccHerokuAddonAttachment_Named
=== RUN   TestAccHerokuAddonAttachment_Named
--- PASS: TestAccHerokuAddonAttachment_Named (5.90s)
PASS
ok      github.com/terraform-providers/terraform-provider-heroku/heroku 5.911s
```

Passes tests :)
